touchpad/main.c

```diff
@@ -878,44 +903,22 @@ int CALLBACK wWinMain(_In_ HINSTANCE hInstance, _In_opt_ HINSTANCE hPrevInstance
 
     if (g_app_state->call_block_input_flag != 0)
     {
-      // TODO find a way to call BlockInput without the need of UAC permission as KeyFreeze
-      // block_input_retval = BlockInput(TRUE);
-      // printf(FG_YELLOW);
-      // printf("BlockInput(TRUE): %d\n", block_input_retval);
-      // printf(RESET_COLOR);
-      // if (block_input_retval == 0)
-      //{
-      //  mGetLastError();
-      //}
-
-      RECT window_rect = g_app_state->window_rect;
-
-      LONG center_x = window_rect.left + ((window_rect.right - window_rect.left) / 2);
-      LONG center_y = window_rect.top + ((window_rect.bottom - window_rect.top) / 2);
-
-      printf(FG_BRIGHT_BLUE);
-      printf("[window_rect] left: %d, right: %d, top: %d, bottom: %d\n", window_rect.left, window_rect.right, window_rect.top, window_rect.bottom);
-      printf("[caculated_center] x: %d, y: %d\n", center_x, center_y);
-      printf(RESET_COLOR);
-
-      RECT lockCursorRect = (RECT){.left = center_x, .top = center_y, .right = center_x + 1, .bottom = center_y + 1};
-      ClipCursor(&lockCursorRect);
+      if (llMouseHookProc == NULL)
+      {
+        llMouseHookProc = SetWindowsHookEx(WH_MOUSE_LL, hHook, NULL, 0);
+      }
 
       g_app_state->call_block_input_flag = 0;
     }
     else if (g_app_state->call_unblock_input_flag != 0)
     {
-      // TODO find a way to detect keyboard state while blocking input
-      // block_input_retval = BlockInput(FALSE);
-      // printf(FG_YELLOW);
-      // printf("BlockInput(FALSE): %d\n", block_input_retval);
-      // printf(RESET_COLOR);
-      // if (block_input_retval == 0)
-      //{
-      //  mGetLastError();
-      //}
-
-      ClipCursor(NULL);
+      if (llMouseHookProc != NULL)
+      {
+        // TODO how to unhook?
+        UnhookWindowsHookEx(hHook);
```

Comment:
@alvinhochun I think I got something wrong with the hooking flow. I was able to block mouse input system-wide but I can't unhook it when done.

Reply:
What is that `HHOOK` parameter that we need to pass into `UnhookWindowsHookEx`? Where do we get it?

Reply:
You are confusing the two of `HOOKPROC` and `HHOOK`. `HHOOK` is the value returned from `SetWindowsHookEx` and `HOOKPROC` is the function pointer to your hook function. They are both defined as `void *` which is why your code typechecks, but it's definitely not correct.

Reply:
I got it.